\documentclass{beamer}
\usepackage{graphicx}
\input{../s20xPreambleRBM.tex}

\begin{document}
\newcommand{\thechapter}{11}

<<RC-H11-000, echo=FALSE,warning=FALSE>>=
source("../s20xNotesHelper.R")
## these are global knitr options and settings for the
## whole document
library(knitr)
library(s20x)
## comment = NA removes ## from all output lines
## prompt = TRUE means the console input prompt > is displayed
## tidy = TRUE means the code is properly spaced and tidied.
opts_chunk$set(comment = NA, size = "scriptsize", prompt = TRUE, tidy = TRUE)
#The next line is required for multicomp to work in R4.0.0
#In future, multicomp may be updated and then this will no longer be required.
#options(stringsAsFactors = TRUE) 
@

\title{Chapter \thechapter: \\ Linear models with a single factor explanatory variable having three or more levels \\ (One-way analysis of variance)}

\institute{University of Auckland}


\begin{frame}
\titlepage
\end{frame}


\begin{frame}[t]
\frametitle{Learning Outcomes}
In this chapter you will learn about:
\begin{center}
\vspace{16pt}
  \begin{itemize}
  \item Fitting a model with a single explanatory factor variable with multiple levels, a.k.a., one-way analysis of variance (ANOVA)\footnote{{\bf NOTE:} When people use the term ANOVA (Analysis of Variance),
they are referring to a linear regression model in which all the explanatory variable are factors.}
  \item Interpretting the fitted model
  \item Multiple pairwise comparisons of means
  \item Using \rcode{emmeans} to solve the multiple comparisons problem 
  \item Relevant \rcode{R}-code
  \item Alternative parameterizations of the one-way ANOVA model\footnote{Optional section.}
  \end{itemize}
\end{center}

\end{frame}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\BeginSection{Example with a 5-level explanatory factor variable }
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


% \begin{frame}
% \begin{center}
% {\huge Explanatory factor with multiple levels, \\
%  and the multi comparisons problem}
% \end{center}
% \end{frame}


\begin{frame}[fragile]
% \frametitle{Explanatory factor with many levels ($>2$)}
% \framesubtitle{Fruit fly example}
\frametitle{Example -- Fruit flies}

In this case study we look at how the male fruit-fly's longevity is related
to his reproductive activity.

\begin{center}
\includegraphics[scale=0.9]{Pictures/rfly}
\end{center}

Fruit flies are a very commonly used animal for laboratory experiments because they are easy to maintain and breed. Their short lifespan allows several generations to be observed within a few months. They also have a genome that is very close to that of humans with many genes discovered in humans also found in fruit flies.\footnote{
See \url{https://www.yourgenome.org/facts/why-use-the-fly-in-research} for more background on research with fruit flies.}
\medskip

Previous studies have shown that the longevity (life span) of female fruit flies decreases with an increase in reproduction, and this leads to a similar question related to males.
\end{frame}


\begin{frame}[fragile]
\frametitle{Fruit fly}
The experiment compared the lifespan of males that were divided into 5 treatment groups that varied according to the presence or absence and number of uninterested or interested females.\footnote{Had there been only two treatment groups then we could have used the two sample two-sample \ttest{} discussed in Chapter 5}
\medskip

How does one define ``interest" in female fruit flies? Here is this study's definition:
\begin{itemize}
\item [] Newly inseminated females will not usually mate again for at least two days.
So, the males in the uninterested females treatment groups were always living
with newly inseminated females.
\end{itemize}
\bigskip

The {\bf primary focus of this Example} is the following: Due to the explanatory factor variable (treatment group) having several levels, we will need to apply an adjustment to relevant \pval{s} and confidence intervals when we are making inference about differences in the expected lifespans between pairs of treatment groups.
\end{frame}



\begin{frame}[fragile]
\frametitle{Fruit fly\ldots}

The response variable measured was \rcode{days}, the number of days the male fly lived.
\bigskip
  

The explanatory factor variable was \rcode{group}, with five levels:

\begin{itemize}
 \item  \rcode{G1} males living alone, \\
 \item  \rcode{G2} males living with one interested female, \\
 \item  \rcode{G3} males living with eight interested females,\\
 \item  \rcode{G4} males living with one uninterested female, and \\
 \item  \rcode{G5} males living with eight uninterested females.
\end{itemize}
\medskip

There were 25 male flies in each group, for a total sample size of 125.
\end{frame}


\begin{frame}[fragile]
\frametitle{Fruit fly\ldots}

Let us take a look at the data:
<<RC-H11-001, fig.show = 'hide'>>=
Fruitfly.df = read.csv("Data/Fruitfly.csv", header=T)
Fruitfly.df$group=factor(Fruitfly.df$group)
boxplot(days ~ group, data = Fruitfly.df, ylab = "Longevity (days)")
@

<<RC-H11-002, echo = FALSE>>=
trimPlot(days ~ group,
         data = Fruitfly.df,
         fileName = "figure/RC-H11-002.pdf",
         plotCommand = boxplot,
         .at = c(1, 2, 3, 4, 5),
         .labels = c("G1", "G2", "G3", "G4", "G5"),
         fig.height = 2.2,
         fig.width = 4,
         x.lab = "Group",
         y.lab = "Longevity (days)")
@

\begin{figure}
  \centering
  \includegraphics{figure/RC-H11-002}
\end{figure}

It looks like male fruit flies do not live as long when in the presence of `uninterested' females (\rcode{G5}), especially when there are several of them.
\end{frame}



\begin{frame}[fragile, label={cnrpt_model}]
\frametitle{Fruit fly\ldots}
\framesubtitle{Linear  model with multi-level ($>2$) explanatory factor}

As seen in previous chapters that involved categorical explanatory variables, our model specification uses indicator variables. In this case:

\[\rcode{days}=\beta_0+\beta_{1}\times\rcode{D2} +\beta_{2}\times\rcode{D3} + \beta_{3}\times\rcode{D4}+  \beta_{4}\times\rcode{D5} + \epsilon\]
where, as usual $\epsilon \iid N(0, \sigma^2)$, and

\medskip
\begin{itemize}
\item \rcode{D2} is an indicator variable whereby \rcode{D2}=1 if the fruit fly is in group 2, otherwise it is 0.

\item \rcode{D3} is an indicator variable whereby \rcode{D3}=1 if the fruit fly is in group 3, otherwise it is 0.

\item ... and so on.
\end{itemize}

For example, $\beta_{1}$ and $\beta_{2}$ represent the differences in expected longevity (\rcode{days}) when we compare groups 2 and 3 to group 1 (the baseline).

%Take a look at the design matrix (5 groups of 4 obs each)
%Grp=rep(c("G1","G2","G3","G4","G5"),c(4,4,4,4,4))
%y=rnorm(20,60,10)
%MadeUpDataFrame=data.frame(Grp,y)
%lmFit=lm(y~Grp,data=MadeUpDataFrame)
%model.matrix(~Grp,data=MadeUpDataFrame)
\end{frame}


\begin{frame}[fragile]
\frametitle{Fruit fly\ldots}
\framesubtitle{Assumption checks}

% Let us see how this model works out with the usual assumption checks.
<<RC-H11-003, fig.show = 'hide'>>=
Fruitfly.fit = lm(days ~ group, data = Fruitfly.df)
plot(Fruitfly.fit, which=1)
@

<<RC-H11-004, echo = FALSE>>=
trimPlot(Fruitfly.fit,
         fileName = "figure/RC-H11-004.pdf",
         which=1,
         fig.height = 2.25,
         fig.width = 4.25,
         cex = 0.7,
         mai = c(0.5, 0.5, 0.1, 0.1),
         x.lab = "Fitted values",
         y.lab = "Residuals")
@

\begin{figure}
  \centering
  \includegraphics{figure/RC-H11-004}
\end{figure}

The {\bf EOV} assumption seem to be okay.
\end{frame}


\begin{frame}[fragile]
\frametitle{Fruit fly\ldots}
\framesubtitle{Assumption checks\ldots}

<<RC-H11-005, fig.show = 'hide'>>=
normcheck(Fruitfly.fit)
@

<<RC-H11-006, echo=FALSE>>=
trimPlot(Fruitfly.fit,
         fileName = "figure/RC-H11-006.pdf",
         plotCommand = normcheck,
         fig.height = 2.1,
         fig.width = 4.1)
@

\begin{figure}
  \centering
  \includegraphics[scale = 0.5]{figure/RC-H11-006}
\end{figure}

The normality assumption seem to be okay.
\end{frame}


\begin{frame}[fragile]

\frametitle{Fruit fly\ldots}
\framesubtitle{Assumption checks\ldots}

<<RC-H11-007, fig.show = 'hide'>>=
cooks20x(Fruitfly.fit)
@

<<RC-H11-008, echo = FALSE, fig.align = 'center', fig.height = 2.5, fig.width = 5>>=
trimPlot(Fruitfly.fit,
         fileName = "figure/RC-H11-008.pdf",
         plotCommand = cooks20x,
         fig.height = 2.1,
         fig.width = 4.1,
         mai = c(0.5, 0.6, 0.1, 0.1))
@

\begin{figure}
  \centering
  \includegraphics[scale = 0.5]{figure/RC-H11-008}
\end{figure}

No unduly influential data points.

\end{frame}

\begin{frame}[fragile, label={fruitfly_anova}]
\frametitle{Fruit fly\ldots}
\framesubtitle{$R^2$ and ANOVA table}
We can trust the fitted model. What can we conclude?\footnote{Recall from Chapter 9 that we have to use the \rcode{anova} function to check the significance of a factor variable with more than two levels.}
\medskip

<<RC-H11-009, echo = 1>>=
anova(Fruitfly.fit)

anova.table <- anova(Fruitfly.fit)
groupSS <- sprintf("%5.0f", anova.table$`Sum Sq`[1])
residualSS <- sprintf("%5.0f", anova.table$`Sum Sq`[2])
totalSS <- sprintf("%5.0f", anova.table$`Sum Sq`[1] + anova.table$`Sum Sq`[2])
RSq <- signif(summary(Fruitfly.fit)$r.squared * 100, 2)
@

\medskip

This allows us to say that there is very strong evidence of a difference in expected longevity between the five groups, which was fairly obvious from the boxplot.
\smallskip

A significant result means we should now investigate how the groups differ from one another -- there is more work to be done.

\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\BeginSection{Interpreting the output}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{frame}[fragile,label=refcell_coeffs]
\frametitle{Fruit fly\ldots}
\framesubtitle{Interpretation}
Now we know that the variable \rcode{group} helps explain longevity,
what can we say about these groups? Let us investigate.

<<RC-H11-010, results='hide'>>=
summary(Fruitfly.fit)
@

<<RC-H11-011, echo=FALSE>>=
slimSummary(Fruitfly.fit)
@

Our fitted model only gives the pairwise difference between the baseline group \rcode{G1} and the other four groups, and so is only providing partial information. What else can we do?
\end{frame}


\begin{frame}[fragile]
\frametitle{Fruit fly\ldots}
\framesubtitle{Interpretation of grand and group means}
Some researchers like to examine the group means and their
deviations from the overall (or so-called``grand'') mean.\footnote{See the optional final Section of this Chapter for more on this topic.} 
These deviations are commonly called group ``effects''.
\medskip

The estimated grand mean is simply the sample mean over all 125 male flies:
<<RC-H11-012a>>=
grand.mean=mean(Fruitfly.df$days)
grand.mean
@

The estimated group means are just the sample means within each group. We can quickly obtain these using the incredibly useful\footnote{\rcode{dplyr} and its associated packages are widely used for ``data wrangling'' (the process of cleaning and re-arranging data sets for easy access and analysis).} \rcode{dplyr} package:


<<RC-H11-012b, warning=F, message=F>>=
library(dplyr()) 
Df=Fruitfly.df %>% group_by(group) %>% summarize(group.mean=mean(days)) %>% 
  data.frame()
@

\end{frame}



\begin{frame}[fragile, label={group_means}]
\frametitle{Fruit fly\ldots}
\framesubtitle{Interpretation of grand and group means}
The above code groups the data by \rcode{group} and then applies the \rcode{mean} function to the \rcode{days} values within each group.
\medskip

The estimated group means are
<<RC-H11-012c>>=
Df$group.mean
@
and the estimated group effects are therefore
<<RC-H11-012d>>=
Df$group.mean-grand.mean
@
\medskip

We have seen that the overall average longevity of the 125 male flies in the study
is about 57.4 days.
\medskip 

We also see that group \rcode{G5} has markedly lower longevity (18.72 fewer days) compared to the overall mean.
\bigskip

We could test null hypotheses and calculate confidence intervals for the above conclusions, but our focus on this course will be making inference about the differences in group means.
\end{frame}



\begin{frame}[fragile]
\frametitle{Fruit fly\ldots}
\framesubtitle{Pairwise comparisons}
We'd really like to get the pairwise comparisons between every possible pair of groups.
However, we've seen that the fitted model is restricted to examining how the groups \rcode{G2-G5} differ from  the baseline group \rcode{G1}.
\medskip

If we wish to see how the other groups differed from group \rcode{G2}, say, then we could achieve this by changing the baseline group to group \rcode{G2}. Recall that this can be done using the \rcode{relevel} function:

<<RC-H11-013, tidy = FALSE>>=
Fruitfly.df$newgroup = relevel(Fruitfly.df$group, ref="G2")
@

But to get all pairwise comparisons (i.e., \rcode{G3~vs~G4}, \rcode{G4~vs~G5}, \rcode{\ldots}) we have to do this re-leveling for \rcode{G2}, \rcode{G3} and \rcode{G4}, and refit the model each time. This is too tedious.
\medskip

We can get \rcode{R} to do the `heavy lifting' for us by using the \rcode{emmeans} function from the \rcode{R} package of the same name. Moreover, \rcode{emmeans} solves the multiple comparisons problem that is discussed below.

\end{frame}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\BeginSection{The multiple comparisons problem}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{frame}[fragile]
\frametitle{Fruit fly\ldots}
\framesubtitle{Multiple comparisons}

Note that when we are looking at all pair-wise comparisons of 5 groups,
we have a total of 10 different possibilities: \medskip

\rcode{G1~vs~G2}, \rcode{G1~vs~G3}, \rcode{G1~vs~G4}, \rcode{G1~vs~G5}, (4 comparisons)

\rcode{G2~vs~G3}, \rcode{G2~vs~G4}, \rcode{G2~vs~G5}, (3 comparisons)

\rcode{G3~vs~G4}, \rcode{G3~vs~G5}, (2 comparisons)

\rcode{G4~vs~G5}, (1 comparisons).

\medskip
In general, if there are $m$ groups then there are $^{m}C_2$ possible pairwise comparisons.\footnote{In \rcode{R} this is given by \rcode{choose(m,2)} and is the number of ways of choosing 2 objects from $m$ objects. E.g., \rcode{choose(5,2)=10}.}

\medskip
Each comparison requires a hypothesis test for a significant difference and an accompanying confidence interval. The multiple comparisons problem arises because, of all null hypotheses that are true, 5\% are falsely rejected (Type 1 error). Equivalently, of all 95\% confidence intervals, 5\% of them do not contain the true parameter value.

\end{frame}



\begin{frame}[fragile]
\frametitle{Erroneous evidence of an effect from multiple testing}
\framesubtitle{Multiple comparisons\ldots}
The following \rcode{R} code fits a simple linear regression model to iid (independent and identically distributed) normal data.  

\medskip

{\bf NOTE:} The null hypothesis $H_0:$ slope = 0 is {\bf true}. 

<<RC-H11-014>>=
x = 1:30 ## Our explanatory variable
x
y = rnorm(30) ## y has NO relationship with x
summary(lm(y~x))$coef ## Print only the coefficient table
@

If this code is run many times over, then approximately 5\% of the time the slope will have \pval{} $<0.05$.\footnote{In fact, it can be shown that the \pval{} is uniformly distributed between 0 and 1 when $H_0$ is true.} \\
\smallskip
That is, there will be erroneous evidence of an effect of \rcode{x} (i.e., evidence for a non-zero slope) about 1 time in 20!

\end{frame}


\begin{frame}[fragile]
\frametitle{Erroneous evidence of an effect from multiple testing\ldots}
\framesubtitle{Multiple comparisons\ldots}

When we do multiple tests (i.e., the 10 paired comparisons in this example) then we greatly increase the probability of obtaining at least one erroneous conclusion\footnote{ Assuming independent comparisons, if we do 10 95\% CIs we have an overall error rate of $1-(1-.05)^{10}=40\%$, which is much higher than our original 5\% error rate per comparison.}.

\medskip 

This is known as the multiple comparison problem. It essentially says that if you look at enough things you will find something `happening', even when there's nothing going on.

\medskip

Remember, data always have variability, and if we are not careful we can `discover' false structure that is not really there.

\medskip 

So, when we look at these 10 comparisons we need to adjust so that the overall error rate (the probability of any spurious significance) over all 10 comparison is no more the 5\%. This can be done using a Tukey adjustment.

\end{frame}


\begin{frame}[fragile]
\frametitle{Example---Fruit fly}
\framesubtitle{Tukey simultaneous confidence intervals}
% \framesubtitle{Multiple comparisons\ldots}

Let's get {\em simultaneous} 95\% confidence intervals for all 10 comparisons via the \rcode{pairs} and \rcode{emmeans} functions of the \rcode{emmeans} package.\footnote{These confidence intervals are called ``simultaneous'' since we can be 95\% confident that {\bf they all} contain the true group difference simultaneously.}

%<<RC-H11-015, size = "scriptsize",warning=F>>=
%library(emmeans)
%Fruitfly.emm = emmeans(Fruitfly.fit, specs = pairwise~group, infer=T)
%Fruitfly.emm$contrasts
%@

<<RC-H11-015, size = "scriptsize",warning=F>>=
library(emmeans)
Fruitfly.pairs = pairs(emmeans(Fruitfly.fit, ~group, infer=T))
Fruitfly.pairs
@
\end{frame}


\begin{frame}[fragile]
\frametitle{Fruit fly}
\framesubtitle{Tukey simultaneous confidence intervals\ldots}

We see that the majority of these pairwise comparisons are not significantly different. Let's extract only the CIs where the Tukey adjusted \pval{}s are less than 0.05.

\medskip
  
<<RC-H11-017>>=
  Fruitfly.pairs=data.frame(Fruitfly.pairs)
  ## Which pairwise comparisons have a P-value less than 0.05?
  mc.signif = Fruitfly.pairs[,"p.value"] < 0.05
  mc.signif
  ## Print comparisons which have a P-value less than 0.05
  print(Fruitfly.pairs[mc.signif, ], digits = 4)
@

{\bf Note} the use of the \rcode{data.frame} function in the above code. We needed to convert \rcode{Fruitfly.pairs} to a dataframe before we could take the subset of rows, otherwise \rcode{emmeans} gets confused and thinks we are doing a smaller number of pairwise comparisons. 

\end{frame}
  

\begin{frame}[fragile]
\frametitle{Fruit fly\ldots}
% \framesubtitle{Tukey simultaneous confidence intervals\ldots}

Some conclusions:
\begin{itemize}
  \item Our model explains 31\% of variability in fruit fly longevity.

  \item We see that group 5 (males with 8 uninterested females) is different from all the others. 

\end{itemize}

On average, group 5 males live fewer days than:

\begin{itemize}

\item Group 1 (males living alone) by 13 to 36 fewer days.

\item Group 2 (males living with one interested female) by 14 to 38 fewer days.

\item Group 3 (males living with eight interested females) by 13 to 36 fewer days.

\item Group 4 (males living with one uninterested female) by 6 to 30 fewer days.
\end{itemize}

\end{frame}


\begin{frame}[fragile]
\frametitle{Fruit fly\ldots}

On a lighter note there is little evidence of a difference in longevity
if no females or only one uninterested female is about,
or if females are there and 'interested' in them ---
but in the presence of multiple uninterested females they die earlier
(they 'drop like flies').

\medskip

Recall also that in the other studies it was seen that females did not live as long if
they reproduced, which can be attributed to the physical demands of producing and laying eggs. With males, perhaps it is sexual frustration that is killing them!

\medskip

For more on this topic see the research article written by Branco et al. (2017, Reproductive activity triggers accelerated male mortality and decreases lifespan: genetic and gene expression determinants in Drosophila. Heredity 118, 221-228 https://doi.org/10.1038/hdy.2016.89) at
\url{https://www.nature.com/articles/hdy201689}.
\end{frame}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\BeginSection{Closing remarks and relevant \rcode{R}-code}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[fragile]
\frametitle{Understanding the \rcode{anova} function output}

<<RC-H11-009b>>=
anova(Fruitfly.fit)
@

In the above output we see that the variability we observe in our longevity data can be broken down into two components \rcode{group} and \rcode{residual}.
\medskip

The amount of variability that the variable \rcode{group} (as shown in the \rcode{Sum Sq} column) explains is $\Sexpr{groupSS}$.  
The residual variability (left over) is $\Sexpr{residualSS}$. 
The total variability is $\Sexpr{groupSS}+\Sexpr{residualSS}=\Sexpr{totalSS}$. 
The \% of variability explained by \rcode{group} is therefore
{\scriptsize
\[
100\times \left(\frac{\Sexpr{groupSS}}{\Sexpr{groupSS}+\Sexpr{residualSS}}\right) =100\times \left(1-\frac{\Sexpr{residualSS}}{\Sexpr{groupSS}+\Sexpr{residualSS}}\right)=\Sexpr{RSq} \%.
\] }

Note that we have just calculated the $R^2$ -- the proportion of the variability in the response variable that is explained by the explanatory variables, 0.31.

\end{frame}



\begin{frame}[fragile]
\frametitle{Most of the \rcode{R}-code you need for this chapter}

Use box plots to inspect the data for each level of the factor.

<<RC-H11-19, eval=F, comment=NA>>= 
boxplot(days ~ group, data = Fruitfly.df)
@
\medskip

You do not need to create indicator variables - \rcode{R} does that for you.
The baseline can be changed if you wish by using the \rcode{relevel} function.

<<RC-H11-020, eval=F, comment=NA>>= 
Fruitfly.df$newgroup = relevel(Fruitfly.df$group, ref="G2")
@

Fit the model and use the ANOVA table to see if any of the means differ from one another (regardless of the baseline chosen).

<<RC-H11-20, eval=F, comment=NA>>=
anova(Fruitfly.fit)
@

Adjust confidence intervals for multiple pairwise comparisons by using the Tukey adjustment to obtain simultaneous intervals CIs:

<<RC-H11-21, eval=F, comment=NA>>=
Fruitfly.pairs = pairs(emmeans(Fruitfly.fit, ~group, infer=T))
@

\end{frame}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\BeginSection{Alternative parameterizations of the 1-way ANOVA model \\ ~ \\ 
(This is an optional Section: \\ - your lecturer will advise whether it is examinable)}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{frame}[fragile]
\frametitle{Alternative parameterizations of the 1-way ANOVA model}
\framesubtitle{The reference cell model}
  
Recall the linear model\footnote{See slide~\ref{cnrpt_model}.} we used to represent the longevity, in days, of a male fruitfly, i.e.
  
\vspace{-2mm}
  
\[\rcode{days}=\beta_0+\beta_{1}\times\rcode{D2} +\beta_{2}\times\rcode{D3} + \beta_{3}\times\rcode{D4}+  \beta_{4}\times\rcode{D5} + \epsilon\]
  
\vspace{-5mm}
  
The {\em parameters} $\beta_0,\beta_1,\ldots,\beta_4$ denote the true values of some attribute (e.g. longevity) of the population of male fruitflies. Here, $\beta_0$ represents the mean longevity of male fruitflies in group \rcode{G1}. The parameters $\beta_1,\ldots,\beta_4$ represent the deviations in mean longevity of males in groups \rcode{G2,\ldots,G5}, respectively, from the mean longevity of males in group \rcode{G1}. 
  
\medskip
  
The values in the \rcode{Estimate} column of the regression summary table\footnote{See slide~\ref{refcell_coeffs}; Coefficients rounded to 2 decimal places.} result in the following equation for predicted longevity:
    
<<RC-H11-003-1, echo=FALSE>>=
beta0 = sprintf("%.2f", coef(Fruitfly.fit)[1])
beta1 = sprintf("%.2f", coef(Fruitfly.fit)[2])
beta2 = sprintf("%.2f", coef(Fruitfly.fit)[3])
beta3 = sprintf("%.2f", coef(Fruitfly.fit)[4])
beta4 = sprintf("%.2f", coef(Fruitfly.fit)[5])
@
    
\medskip
  
\begin{center}
\small
$ \widehat{\rcode{days}}=\Sexpr{beta0}+\Sexpr{beta1}\times \rcode{D2} + (\Sexpr{beta2})\times \rcode{D3} + (\Sexpr{beta3})\times \rcode{D4} + (\Sexpr{beta4})\times \rcode{D5}$
\end{center}
  
\end{frame}
  

\begin{frame}[fragile]
\frametitle{Alternative parameterizations of the linear model}
\framesubtitle{The reference cell model}
  
\small

Each cell within a column in the table below corresponds to a level of the \rcode{Group} factor. One way to `parametrise' these cells is to use means, i.e. $\mu_1,\mu_2,\ldots,\mu_5$. Another is to select one of the cells as a reference cell (here \rcode{Group G1}) and the remaining cells are then parametrised the deviations of the current row's group mean from the reference cell's group mean.
  
\medskip

\begin{center}
\footnotesize
\renewcommand{\arraystretch}{1.15}
   \begin{tabular}{ccccccrr}
      \hline
      &&& \multicolumn{5}{c}{parameterization} \\
      \cline{4-8}
      Group & Data && Means & Estimate\footnote{See estimates of \rcode{Group} means on slide~\ref{group_means}} &&
         Reference cell & Estimate\footnote{See regression coefficients table on slide\ref{refcell_coeffs}} \\
      \hline
      \rcode{G1} & $40,37,\ldots,44$ && $\mu_1$ & 63.56 && $\beta_0 = \mu_1$ & 63.56 \\
      \rcode{G2} & $46,42,\ldots,92$ && $\mu_2$ & 64.80 && $\beta_1 = \mu_2 - \mu_1$ & 1.24 \\
      \rcode{G3} & $35,37,\ldots,77$ && $\mu_3$ & 63.36 && $\beta_2 = \mu_3 - \mu_1$ & $-0.20$  \\
      \rcode{G4} & $21,40,\ldots,68$ && $\mu_4$ & 56.76 && $\beta_3 = \mu_4 - \mu_1$ & $-6.80$ \\
      \rcode{G5} & $16,19,\ldots,44$ && $\mu_5$ & 38.72 && $\beta_4 = \mu_5 - \mu_1$ & $-24.84$  \\
      \hline
   \end{tabular}
\end{center}

\medskip

The parameterization of the model shown on the previous slide is therefore known as the {\em reference cell} model. 

\end{frame}
  

\begin{frame}[fragile]
\frametitle{Alternative parameterizations of the linear model}
\framesubtitle{The means model}
  
From the above table we can see that there is an alternative, but equivalent, {\em means} model parameterization, i.e. linear model for the longevity of the $j$th ($j=1,2,\ldots,25$) male fruitfly in \rcode{Group} $i$ ($i=1,2,\ldots,5$) may be written as

\medskip

\[ days_{ij} = \mu_i + \epsilon_{ij} \]

where $\mu_i$ denotes the mean longevity, in days, of a male in \rcode{Group} $i$ and, as usual, $\epsilon_{ij} \iid N(0, \sigma^2)$.

\end{frame}


\begin{frame}[fragile]
\frametitle{Alternative parameterizations of the linear model}
\framesubtitle{The effects model}

\small

Another parameterization is to set the overall mean longevity, $\mu$, as the reference and then define the {\em effect}, $\tau_i$, on longevity due to being in \rcode{Group} $i$ as the difference between the  \rcode{Group} $i$ mean and the overall mean, i.e. $ \tau_i = \mu_i - \mu $.

\medskip

\begin{center}
\footnotesize
\renewcommand{\arraystretch}{1.15}
   \begin{tabular}{ccccccrr}
      \hline
      &&& \multicolumn{5}{c}{parameterization} \\
      \cline{4-8}
      Group & Data && Means & Estimate &&
         Effects & Estimate\footnote{See overall mean (57.44 days) and deviations of group means from overall means on slide~\ref{group_means}.} \\
      \hline
      \rcode{G1} & $40,37,\ldots,44$ && $\mu_1$ & 63.56 && $\tau_1 = \mu_1 - \mu$ & 6.12 \\
      \rcode{G2} & $46,42,\ldots,92$ && $\mu_2$ & 64.80 && $\tau_2 = \mu_2 - \mu$ & 7.36 \\
      \rcode{G3} & $35,37,\ldots,77$ && $\mu_3$ & 63.36 && $\tau_3 = \mu_3 - \mu$ & 5.92  \\
      \rcode{G4} & $21,40,\ldots,68$ && $\mu_4$ & 56.76 && $\tau_4 = \mu_4 - \mu$ & $-0.68$ \\
      \rcode{G5} & $16,19,\ldots,44$ && $\mu_5$ & 38.72 && $\tau_5 = \mu_5 - \mu$ & $-18.72$  \\
      \hline
   \end{tabular}
\end{center}

\medskip
  
The linear {\em effects} model for the longevity of the $j$th ($j=1,2,\ldots,25$) male fruitfly in \rcode{Group} $i$ ($i=1,2,\ldots,5$) may therefore be written as

\vspace{-4mm}

\[ days_{ij} = \mu + \tau_i + \epsilon_{ij} \]

\vspace{-5mm}

where, again, $\epsilon_{ij} \iid N(0, \sigma^2)$.

\end{frame}
  

\end{document}